finetune/util/input_utils.py

```diff
@@ -61,21 +58,72 @@ def add_length(x, y=None):
     return x
 
 
-def batch_dataset(dataset, batch_size, shapes, n_epochs=1, shuffle=False):
+def batch_dataset(
+    dataset,
+    batch_size,
+    shapes,
+    max_length,
+    n_epochs=1,
+    shuffle=False,
+    table_batching=False,
+    random_seed=42,
+):
     if isinstance(shapes, tuple):
         shapes = ({**shapes[0], "length": tf.TensorShape([])}, shapes[1])
     else:
         shapes = {**shapes, "length": tf.TensorShape([])}
 
-    def batched_dataset():
-        return (
-            dataset()
-            .map(add_length)
-            .shuffle(500 if shuffle else 1)
-            .padded_batch(batch_size, padded_shapes=shapes, drop_remainder=False)
-            .repeat(n_epochs)
-            .prefetch(tf.data.experimental.AUTOTUNE)
-        )
+    if table_batching:
```

Comment:
would be better as an overridden method somewhere, but I don't think base-model is the right place to put it and it wouldn't work on input Pipeline as this is base-model specific. I'll probably leave this here for now unless someone has a better idea.